cwlVersion: v1.2
class: CommandLineTool

label: "VirSorter2 workflow"
doc: |
  This CWL runs VirSorter2 for identifying viral DNA and RNA sequences from metagenomic assemblies.

requirements:
  - class: DockerRequirement
    dockerPull: quay.io/biocontainers/virsorter:2.2.4--pyhdfd78af_1

  - class: ResourceRequirement
    coresMin: 2
    coreMax: 30

inputs:
  input_fasta:
    type: File
    label: Input FASTA File
    doc: The metagenomic assembly in FASTA format.
    inputBinding:
      position: 1

  db_dir:
    type: Directory
    label: Database Directory
    doc: Directory containing the VirSorter2 database.
    inputBinding:
      position: 2

  threads:
    type: int
    label: Number of Threads
    doc: Number of threads to use for VirSorter2.
    inputBinding:
      position: 3
    default: 4

outputs:
  result_folder:
    type: Directory
    label: Result Folder
    doc: The output folder generated by VirSorter2.
    outputBinding:
      glob: "*_resFolder"

baseCommand: [ "virsorter", "run" ]

arguments:
  - valueFrom: bash
    position: 0

  - prefix: --input-fasta
    valueFrom: $(inputs.input_fasta.path)
    position: 1

  - prefix: --db
    valueFrom: $(inputs.db_dir.path)
    position: 2

  - prefix: --threads
    valueFrom: $(inputs.threads)
    position: 3

  - prefix: --working-dir
    valueFrom: ./virsorter_results
    position: 4
